Source/Schema.NET/OneOrMany{T}.cs

```diff
@@ -278,11 +278,11 @@ public bool Equals(OneOrMany<T> other)
             }
             else if (this.HasOne && other.HasOne)
             {
-                return this.collection[0].Equals(other.collection[0]);
+                return Equals(this.collection![0], other.collection![0]);

```

Comment:
Was the reason for changing to use `object.Equals(self, other)` to avoid checking directly if `this.collection[0]` is `null`?

Reply:
Yes. The alternative would be to check all the combinations `null==null`, `null==object`, `object==null` and `object==object`.